tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/GenerateRepositoryFacadeMojo.java

```diff
@@ -520,6 +522,30 @@ private void alterContentJar(File p2repository) throws FileNotFoundException, IO
 		contentStream.close();
 		outContentStream.closeEntry();
 		outContentStream.close();
+
+		// JBDS-3929 overwrite the content.xml.xz file too
+		// see also https://bugs.eclipse.org/bugs/show_bug.cgi?id=464614
+		//getLog().debug("delete old content.xml.xz");
+		File contentXmlXz = new File(p2repository,"content.xml.xz");
+		FileUtils.forceDelete(contentXmlXz);
+		//getLog().debug("create content.xml from transformed XML");
+		File contentXml = new File(p2repository, "content.xml");
+		FileOutputStream outContentStreamXml = new FileOutputStream(contentXml);
+		StreamResult resultXml = new StreamResult(outContentStreamXml);
+		transformer.transform(source, resultXml);
+		outContentStreamXml.close();
+		//getLog().debug("stream content.xml to content.xml.xz");
+		BufferedInputStream in = new BufferedInputStream(new FileInputStream(contentXml));
+		XZCompressorOutputStream out = new XZCompressorOutputStream(new FileOutputStream(contentXmlXz));
+		final byte[] buffer = new byte[1024];
+		int n = 0;
+		while (-1 != (n = in.read(buffer))) {
+		    out.write(buffer, 0, n);
+		}
+		out.close();
```

Comment:
extracted method alterXzFile throws IOException, FileNotFoundException, TransformerException ... is that enough or do you still need try/catch blocks to trap for these errors and dump more details to console?